### PR TITLE
Show registration status on My Competitions page

### PR DESCRIPTION
**Pull request by pingskills**
The list and registered icons from Font Awesome Icons have been used
to show the registration status along with some extra text in the tooltip.
Two tests were created to test the my_competitions view.

### CODE REVIEW

**Comment by pingskills** [comment]
Here are some images of what the icons and tootips look like:

Registered tooltip
my-competitions-registered

Waiting List Tooltip
my-competitions-waiting-list

If the user is registered, should I make the icon a link to the competitors page?

File: WcaOnRails/app/views/competitions/_my_competitions_table.html.erb [hunk 1]
```diff
@@ -15,29 +15,47 @@
...
           <% end %>
         <% end %>
           <tr class="<%= competition.isConfirmed ? "confirmed" : "not-confirmed" %> <%= competition.showAtAll ? "visible" : "not-visible" %> <%= !past ? "not-past" : "past" %>" data-toggle="tooltip" data-placement="top" data-container="body" title="<%= !past ? message : "" %>">
             <td><%= link_to competition.name, competition_path(competition) %></td>
             <td><%= competition.cityName %>, <%= competition.country ? competition.country.name : "" %></td>
             <td><%= wca_date_range(competition.start_date, competition.end_date) %></td>
             <td>
+              <% if registration %>
+                <% if registration.accepted? %>
+                  <span class="fa fa-fw fa-registered"></span>
```
Comment (jonatanklosko) [inline WcaOnRails/app/views/competitions/_my_competitions_table.html.erb]: We have a helper for that (comes with font-awesome-sass gem). For example
```ruby
<%= icon("registered") %>
```

File: WcaOnRails/app/views/competitions/_my_competitions_table.html.erb [hunk 2]
```diff
@@ -15,29 +15,47 @@
...
+                <% else %>
+                  <span class="fa fa-fw fa-list"></span>
+                <% end %>
```
Comment (jonatanklosko) [inline WcaOnRails/app/views/competitions/_my_competitions_table.html.erb]: Actually we can simplify the above with
```ruby
<%= icon(registration.accepted? ? "registered" : "list") %>
```

File: WcaOnRails/spec/views/competitions/my_competitions.html.erb_spec.rb [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+  before do
+    allow(view).to receive(:current_user) { registration.user }
+    assign(:not_past_competitions, [competition])
+    assign(:past_competitions, [])
+  end
+
+  it "shows upcoming competitions" do
+    render
+    expect(rendered).to match /<td><a href="\/competitions\/MelbourneOpen2016">Melbourne Open 2016<\/a><\/td>/
```
Comment (jonatanklosko) [inline WcaOnRails/spec/views/competitions/my_competitions.html.erb_spec.rb]: `match` can also test for string so you can use single quotes and avoid escaping with `\/`

Comment (pingskills) [inline WcaOnRails/spec/views/competitions/my_competitions.html.erb_spec.rb]: Rubocop didn't like the escaping and suggested %r which is what I changed it to.

Comment (pingskills) [inline WcaOnRails/spec/views/competitions/my_competitions.html.erb_spec.rb]: I think you're right though, using a string seems more appropriate here.

**Comment by jonatanklosko** [comment]
Hmm, the icons aren't well descriptive in my opinion. What about "calendar-check-o" and "hourglass-half" (or something else)?

**Comment by jfly** [comment]
LGTM! (of course, respond to jonatan's comments as necessary before merging) Thanks for writing a test!

**Comment by jfly** [comment]
Oh, in the future, don't forget to put "fixes #586" in your commits.

**Comment by jonatanklosko** [comment]
Awesome, LGTM =)

**Comment by pingskills** [comment]
OK. Updated the code as suggested. Here's a screenshot of the new icons. What do you think?

my-competitions-new-icons

**Comment by jonatanklosko** [comment]
Also the icons look great for me =)

**Pull request merged**